_posts/multirepo_comments.adoc

```diff
@@ -0,0 +1,96 @@
+
+Later on you realize that two of the components can share some piece of infrastructure code, so you create a library out of it and put that in its own repo.
+
+Then you want some "magic script" to build your entire system from scratch and produce a shipable product, so you write it up. But where do you put it? Doesn't seem to fit in any of the repos you created... So you create a new repo for build stuff.
+
+Without noticing, you've just created ~10 git repositories for a single product. Welcome to MultiRepo Hell.
+
+While the above example sound extreme, I've actually seen this happen more than once. We'll now examine why this is a terrible way to go.
+
+## MultiRepo Hell
+
+There are a lot of consequences to going MultiRepo, so the list below is not complete.
+
+.Versioning
+Unless your APIs are simple, stable and well defined, you're going to need a way to specify which version process A knows how to talk to which version of process B. This means that *every git repo must be versioned in a sensible way*. And while "sensible" sounds simple enough, it is not (and that might deserve its own blog post). In a nutshell, by sensible I mean that you should be able to specify a dependency on some feature, some bugfix, or some API version. This quickly becomes a headache as developers must keep updating their component's version. When they forget to do so, the entire system might be in an inconsistent state, where components do not agree on which version of some other component they want to use.
+
+Multirepo creates a ton of interdependencies that simply do not exist in Monorepo, since in monorepo there is only one state of the code - the current commit - and not some combinatorial monster that depends on the exact state of each repo.
+
+.Build-Time Dependencies
+In many software projects (especially in C/C++) it is common to have multiple processes share code in the form of a library. This often means that you must build the librbary _before_ you build the dependent process (this is especially true when using static libraries). As a consequence, your upper level build script (which resides in the so called "build" repo) must know in which order to build the components. Moreover, a dependent component must know the relative location of the component it depends on, and it must rely on that component to be built before itself. Yuck.
+
+In Monorepo, relative locations between components are well known - each one has a directory under the root directory. That simplifies relative path issues between components ("where the heck is that `include` folder?"), and makes the upper level build script much simpler.
+
```

Comment:
Loooong, and not a well defined problem

Also, I do agree, but I could argue an say that's a wanted behavior and gives freedom to the developers and reusing parts of the code